backend/src/sysupport/token.js

```diff
@@ -0,0 +1,68 @@
+import config from '../config'
+import rp from 'request-promise'
+
+export async function timedLogin() {
+  const { host, user, password, prefix, authPath } = config.services.timed
+  const res = await rp({
+    method: 'post',
+    uri: `${host}${prefix}${authPath}`,
+    headers: {
+      accept: 'application/vnd.api+json',
+      'content-type': 'application/vnd.api+json'
+    },
+    body: {
+      data: {
+        type: 'obtain-json-web-tokens',
+        id: null,
+        attributes: {
+          username: user,
+          password
+        }
+      }
+    },
+    json: true
+  })
+
+  return res.data.token
+}
+
+export async function refreshToken(token) {
+  const { host, prefix, authRefresh } = config.services.timed
+  const res = await rp({
+    method: 'post',
+    uri: `${host}${prefix}${authRefresh}`,
+    headers: {
+      accept: 'application/vnd.api+json',
+      'content-type': 'application/vnd.api+json'
+    },
+    body: {
+      data: { type: 'refresh-json-web-tokens', attributes: { token } }
+    },
+    json: true
```

Comment:
You always have to repeat `headers` and `json`. Would it make sense to factor this out into something like 
`rp(addHeaders({ uri: ..., body: ...}))`?

Reply:
I think it is maybe a bit too much optimization. As soon as one header changes you need to modify the function again and so on. Like this it is clear what exactly the request does.

Reply:
But if a header changes, it changes for all the calls, right?

Generally i see your point about too much abstraction though - feel free to leave it like that.